### PR TITLE
Flow fixes

### PR DESCRIPTION
**Pull request by danielmoi**
#### Description

This branch makes some fixes for Flow usage

----
#### Changes

- add flow plugin for eslint
- refactor defaultProps, props, and state Flow declarations
  - use base Flow `Styles` and `DefaultStyles`
  - other components maintain their own defaultProps and props
- remove React.PropTypes >> not needed anymore

----
#### Testing Instructions

_Outline the steps to test or reproduce the PR._

----
#### Screenshots

_(if appropriate, they help the reviewer visualise the changes)_

----
#### Checklist:
- [ ] Tests – Have you added or edited the test cases?
- [x] Rebase – Has this PR been rebased against `master`?

### CODE REVIEW

File: src/components/SelectBase.js [hunk 1]
```diff
@@ -1,147 +1,122 @@
...
-    selectedLength = selected.size;
-  }
+type SelectBaseDefaultProps = {
+  styles: DefaultStyles,
+}
 
-  if (Array.isArray(selected)) {
-    selectedLength = selected.length;
-  }
+class SelectBase extends Component {
```
Comment (anupvarghese) [inline src/components/SelectBase.js]: is it updated to react component to validate default props?

File: src/components/SelectBase.js [hunk 2]
```diff
@@ -1,147 +1,122 @@
...
 import classNames from 'classnames';
 import { Map, List } from 'immutable';
 
-import type { SelectBaseProps } from '../types';
+import type { Options, Selected, Callback, Styles, DefaultStyles } from '../types';
 
-const additionalPropTypes = {
-  onCheck: PropTypes.func.isRequired,
-};
+type SelectBaseProps = {
```
Comment (anupvarghese) [inline src/components/SelectBase.js]: this is a good idea 👍

File: src/components/SelectState.js [hunk 3]
```diff
@@ -1,23 +1,62 @@
...
+  searchTerm: '',
+};
+
+type SelectStateState = {
+  isOpen: boolean,
+  options: Options,
+}
+
+export class SelectStateComponent extends Component {
+  defaultProps: SelectStateDefaultProps;
```
Comment (anupvarghese) [inline src/components/SelectState.js]: I am not sure about the right approach but do you think specifying the types in component as `<propType, defaultPropType, stateType>` is better?

File: src/redux/actions.js [hunk 4]
```diff
@@ -1,30 +1,30 @@
 /* @flow */
 
 import C from './constants';
-import type { FSA } from '../types';
+import type { ActionCreator } from '../types';
 
 
-const addSelect: FSA = (data: Object) => ({
+const addSelect: ActionCreator = (data: Object) => ({
```
Comment (anupvarghese) [inline src/redux/actions.js]: Yeah `ActionCreator` is better than Flux state actions 👍

File: src/components/SelectBase.js (hunk 1 above)
Comment (danielmoi) [inline src/components/SelectBase.js]: Going to need to place the search logic in base component

File: src/components/SelectState.js (hunk 3 above)
Comment (danielmoi) [inline src/components/SelectState.js]: Yeah 😝 Need to figure out best practices. Couldn't get generics to work yet..

Comment (anupvarghese) [inline src/components/SelectState.js]: we will discuss today. I was able to see type errors with generics, lets see whats going wrong

**Pull request merged**